app/ts/schema.ts

```diff
@@ -32,25 +28,98 @@ export const BigIntParser: ParsedValueConfig<funtypes.String, bigint> = {
 	},
 }
 
-export const BigIntHexSchema = funtypes.String.withParser(BigIntParser)
+export const BigIntHex = funtypes.String.withParser(BigIntParser)
+
+export const HexString = funtypes.String.withGuard(isHexString)
+export type HexString = funtypes.Static<typeof HexString>
+
+export function createUnitParser(decimals?: bigint): funtypes.ParsedValue<funtypes.String, HexString>['config'] {
+	return {
+		parse(value) {
+			try {
+				const bigIntAmount = parseUnits(value, decimals)
+				const maybeHexAmount = BigIntHex.serialize(bigIntAmount)
+				return HexString.safeParse(maybeHexAmount)
+			} catch {
+				return { success: false, message: `${value} is not a number.` }
+			}
+		},
+		serialize: funtypes.String.safeParse
+	}
+}
 
-export const TokenSchema = funtypes.Object({
-	chainId: BigIntHexSchema,
+export const AddressParser: funtypes.ParsedValue<funtypes.String, string>['config'] = {
+	parse: value => {
+		const addressString = value.toLowerCase()
+		if (!isAddress(addressString)) return { success: false, message: `${value} is not a valid address string.` }
+		else return { success: true, value: getAddress(addressString) }
+	},
+	serialize: funtypes.String.safeParse
+}
+
+export const EthereumAddress = funtypes.String.withParser(AddressParser).withGuard(isHexString)
+export type EthereumAddress = funtypes.Static<typeof EthereumAddress>
+
+export const QuantityParser: funtypes.ParsedValue<funtypes.String, bigint>['config'] = {
```

Comment:
is there a reason you have quantityparser and bigint parser separately?

Reply:
removing as this is unused in this PR